analyzers/src/SonarAnalyzer.Common/Rules/HardcodedIpAddressBase.cs

```diff
@@ -30,64 +30,86 @@
 
 namespace SonarAnalyzer.Rules
 {
-    public abstract class HardcodedIpAddressBase<TLiteralExpression> : HotspotDiagnosticAnalyzer
+    public abstract class HardcodedIpAddressBase<TSyntaxKind, TLiteralExpression> : HotspotDiagnosticAnalyzer
+        where TSyntaxKind : struct
         where TLiteralExpression : SyntaxNode
     {
         protected const string DiagnosticId = "S1313";
         protected const string MessageFormat = "Make sure using this hardcoded IP address '{0}' is safe here.";
+        private const int IPv4AddressParts  = 4;
+        private const string IPv4Broadcast = "255.255.255.255";
 
-        private static readonly ISet<string> IgnoredVariableNames =
-            new HashSet<string>
+        private readonly string[] ignoredVariableNames =
             {
                 "VERSION",
                 "ASSEMBLY",
             };
 
+        private readonly DiagnosticDescriptor rule;
+
+        protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
+
```

Comment:
I would remove this empty line, as it's a block of abstract properties (easier to read)